src/main/java/seedu/address/logic/parser/SortCommandParser.java

```diff
@@ -0,0 +1,89 @@
+import seedu.address.logic.commands.sort.SortResumeCommand;
+import seedu.address.logic.commands.sort.SortSkillCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.item.Item;
+import seedu.address.model.util.ItemUtil;
+
+/**
+ * Parses input arguments and create a SortCommand object.
+ */
+public class SortCommandParser implements Parser<SortCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments and returns a SortCommand object for execution.
+     */
+    @Override
+    public SortCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_ITEM, PREFIX_SORT_ORDER, PREFIX_REVERSE);
+        if (!arePrefixesPresent(argMultimap, PREFIX_ITEM) || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(Item.MESSAGE_CONSTRAINTS);
+        }
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_SORT_ORDER)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, SortCommand.MESSAGE_USAGE));
+        }
+
+        String itemType = ParserUtil.parseItemType(argMultimap.getValue(PREFIX_ITEM).get());
+        String sortOrder = argMultimap.getValue(PREFIX_SORT_ORDER).get();
+        boolean reverse = ParserUtil.parseReverse(argMultimap.getValue(PREFIX_REVERSE).orElse(null));
+
+        if (!sortOrder.equalsIgnoreCase("name")
+            && !sortOrder.equalsIgnoreCase("time")
+            && !sortOrder.equalsIgnoreCase("level")) {
+            throw new ParseException(SortCommand.MESSAGE_SORT_NOT_EXISTED);
+        }
+
+        switch (itemType) {
+        case ItemUtil.RESUME_ALIAS:
+            if (!sortOrder.equalsIgnoreCase("name")) {
+                throw new ParseException(SortCommand.MESSAGE_INAPPLICABLE_SORT);
+            } else {
+                return new SortResumeCommand(sortOrder, reverse);
+            }
+
+        case ItemUtil.INTERNSHIP_ALIAS:
+            if (sortOrder.equalsIgnoreCase("level")) {
+                throw new ParseException(SortCommand.MESSAGE_INAPPLICABLE_SORT);
+            } else {
+                return new SortInternshipCommand(sortOrder, reverse);
+            }
+
+        case ItemUtil.PROJECT_ALIAS:
+            if (sortOrder.equalsIgnoreCase("level")) {
+                throw new ParseException(SortCommand.MESSAGE_INAPPLICABLE_SORT);
+            } else {
+                return new SortProjectCommand(sortOrder, reverse);
+            }
+
+        case ItemUtil.SKILL_ALIAS:
+            if (sortOrder.equalsIgnoreCase("time")) {
+                throw new ParseException(SortCommand.MESSAGE_INAPPLICABLE_SORT);
+            } else {
+                return new SortSkillCommand(sortOrder, reverse);
+            }
+        default:
+            // TODO: Use a better Exception here
+            throw new ParseException("The item type is not detected! Something is wrong");
```

Comment:
Normally for other cases, code will never reach here bc to pass the parseItemType it must be part of the various items, for this case, it will reach here only if it is i/ note. Would you like to have something like "note" is not an acceptable type for sort?